src/OrleansRuntime/MembershipService/GrainBasedMembershipTable.cs

```diff
@@ -32,6 +33,17 @@ internal class GrainBasedMembershipTable : Grain, IMembershipTable
         private InMemoryMembershipTable table;
         private TraceLogger logger;
 
+        public Task InitializeMembershipTable(GlobalConfiguration config, bool tryInitTableVersion, TraceLogger traceLogger)
+        {
+            return OnActivateAsync();
```

Comment:
no need to call OnActivateAsync here. It will be called automatically before first msg is delivered, which happens to be InitializeMembershipTable. so can just move the logic from OnActivateAsync to here or InitializeMembershipTable empty. Otherwise, the init logic will execute twice.